Redesign pytest

Dwij1704 opened the pull request:
## 📥 Pull Request

**📘 Description**
Added instrumentation tests for the Anthropic, OpenAI  and Haystack. These tests ensure that API calls are tracked and verified using spans.

**🧪 Testing**
To validate these changes, run any integration test. The tests utilize VCR to record API calls and cassettes for replaying them. Snapshots are used to verify the expected outputs.

Dwij1704 committed:
test(openai): Update OpenAI instrumentation test to verify span tracking

Dwij1704 committed:
test(anthropic): Update Anthropic instrumentation test to verify span tracking

Dwij1704 committed:
test(haystack): Add instrumentation test for Haystack OpenAI generator

teocns committed:
Reorganize Session's **kwargs, -video attr

Signed-off-by: Teo <teocns@gmail.com>

Dwij1704 committed:
Merge branch 'redesign' of github.com:AgentOps-AI/agentops into redesign-pytest

Dwij1704 committed:
test: Enhance instrumentation tests with VCR and snapshot assertions

Update Anthropic, Haystack, and OpenAI instrumentation tests to:
- Use VCR for recording API interactions
- Implement snapshot-based span verification
- Improve test robustness by capturing and asserting span details

teocns commented on tests/integration/test_anthropic_instrumentation.py:
```diff
@@ -0,0 +1,41 @@
+        spans_json = json.dumps(spans_data, indent=2)
+        
+        # Verify the spans using snapshot
+        snapshot.assert_match(spans_json, 'anthropic_spans.json')
```
this is not looking right, is it? 

snapshots are supposed to _inline_ override the assertion expression, generating the body of the span after the second run 

1.Before running snapshotter:
assert data = snapshot()

2.After running snapshotter:

assert data = ["snapshotter edited the test file and generated this data"]

Dwij1704 replied:
Initially we would need to create a snapshot manually using --snapshot-update flag in pytest, also it only accepts str or bytes that explains the json format
![image](https://github.com/user-attachments/assets/9566cffb-7d27-419e-afa2-b0820f67be15)

Dwij1704 replied:
Also the 'Snapshot' object is not callable

Dwij1704 committed:
test: Update instrumentation tests with inline snapshots and environment setup

teocns commented on tests/integration/test_anthropic_instrumentation.py:
```diff
@@ -0,0 +1,88 @@
+import os
+os.environ["OPENAI_API_KEY"] = "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+os.environ["ANTHROPIC_API_KEY"] = "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+os.environ["AGENTOPS_API_KEY"] = "00000000-0000-0000-0000-000000000000"
```
use fixtures for these and exclude them from snapshots

teocns commented on tests/integration/test_anthropic_instrumentation.py:
```diff
@@ -0,0 +1,88 @@
+def test_anthropic_instrumentation(exporter):
+    """Test that Anthropic API calls are tracked in spans using inline snapshots"""
+
+    session = agentops.start_session()
```
instead of starting the session manually, use fixture `agentops_session` which already handles session start / end, and requires no further action

Dwij1704 committed:
test: Centralize environment setup for integration tests

Refactor integration test files to:
- Remove hardcoded API key environment variables
- Add a new `mock_env_keys` fixture in conftest.py for consistent environment setup
- Simplify test structure by removing manual session management
- Use the new `mock_env_keys` and `agentops_session` fixtures across tests

Dwij1704 committed:
Updated

Dwij1704 committed:
test: Enhance VCR configuration and sensitive data filtering

Dwij1704 committed:
test: Remove hardcoded cassettes from integration tests

teocns committed:
deps(test): add inline-snapshot

Signed-off-by: Teo <teocns@gmail.com>

teocns committed:
deps(test): add haystack-ai

Signed-off-by: Teo <teocns@gmail.com>

teocns committed:
rethink mocks

Signed-off-by: Teo <teocns@gmail.com>
